ext/standard/array.c

```diff
@@ -1157,22 +1151,13 @@ PHP_FUNCTION(reset)
 PHP_FUNCTION(current)
 {
 	zval *array_zv;
-	zval *entry;
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
 		Z_PARAM_ARRAY_OR_OBJECT(array_zv)
 	ZEND_PARSE_PARAMETERS_END();
 
 	HashTable *array = get_ht_for_iap(array_zv, /* separate */ false);
-	if ((entry = zend_hash_get_current_data(array)) == NULL) {
-		RETURN_FALSE;
-	}
-
-	if (Z_TYPE_P(entry) == IS_INDIRECT) {
-		entry = Z_INDIRECT_P(entry);
-	}
-
-	RETURN_COPY_DEREF(entry);
+	ia_return_current(return_value, array, true);
```

Comment:
This will implicitly advance when the current element is unset. E.g. with a `current(), unset(), current()`. I think it would be reasonable for `current()` not to advance, but to throw if the element is undef.

Reply:
I think that's fair, same for key(), let me fix this.